From 680ed71043ea0c7ffba777531e1c9eabe5c19bca Mon Sep 17 00:00:00 2001
From: Naveen Raj Ranganathan <naver@qti.qualcomm.com>
Date: Wed, 1 Jul 2026 17:27:41 +0530
Subject: [PATCH] rc.common: skip services listed in
 /etc/mem-profile.d/<profile> based on active mem-profile bootargs

Change-Id: I21757058e68d5e895a6e5dfc2894f0a068157b51
Signed-off-by: Naveen Raj Ranganathan <naver@qti.qualcomm.com>
---
 package/base-files/files/etc/rc.common | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/package/base-files/files/etc/rc.common b/package/base-files/files/etc/rc.common
index abf72ea3372f..c0486c161faf 100755
--- a/package/base-files/files/etc/rc.common
+++ b/package/base-files/files/etc/rc.common
@@ -8,6 +8,16 @@ initscript=$1
 action=${2:-help}
 shift 2
 
+get_mem_profile() {
+	[ -r /etc/mem-profile ] && tr -d '\n' < /etc/mem-profile
+}
+
+is_service_disabled() {
+	local profile="$(get_mem_profile)"
+	local svc="$1"
+	grep -qx "$svc" "/etc/mem-profile.d/$profile"
+}
+
 start() {
 	return 0
 }
@@ -135,6 +145,10 @@ extra_command "enabled" "Check if service is started on boot"
 	}
 
 	start() {
+		name="$(basename "${initscript}")"
+		name="${name##[SK][0-9][0-9]}"
+		is_service_disabled "$name" && exit 0
+
 		rc_procd start_service "$@"
 		if eval "type service_started" 2>/dev/null >/dev/null; then
 			service_started
-- 
2.34.1

